app/assets/javascripts/admin/modules/ga4-button-setup.js

```diff
@@ -0,0 +1,27 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+;(function (Modules) {
+  function Ga4ButtonSetup(module) {
+    this.module = module
+  }
+
+  Ga4ButtonSetup.prototype.init = function () {
+    const buttons = this.module.querySelectorAll('button')
+    buttons.forEach((button) => {
+      const event = {
+        event_name: button.type === 'submit' ? 'form_response' : 'navigation',
+        type: 'generic_link',
+        text: button.textContent,
+        section: document.title.split(' - ')[0],
+        action: button.textContent,
+        tool_name: document.title.split(' - ')[0]
+      }
+      if (button.dataset.ga4Event) {
+        Object.assign(event, JSON.parse(button.dataset.ga4Event))
+      }
```

Comment:
My JS is rusty, but something like this is more readable IMO:

```suggestion
      const event = (button.dataset.ga4Event) ? JSON.parse(button.dataset.ga4Event) : {
        event_name: button.type === 'submit' ? 'form_response' : 'navigation',
        type: 'generic_link',
        text: button.textContent,
        section: document.title.split(' - ')[0],
        action: button.textContent,
        tool_name: document.title.split(' - ')[0]
      }
```

Reply:
The `Object.assign` here is merging the two objects so that you're able to override only one part of the `ga4-event` without having to reconstruct the whole thing. The ternary operator wouldn't allow for this.

I've added an extra test to illustrate and make sure we don't lose the merging in future.

Reply:
Ahh makes sense. Thanks for explaining